Bottom: fafc391870a37d33fa7ae72e7da77e9ee8644bd4
Top:    80cea9440dfabb3013b3802c7645bf700e89ccdb
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-07-20 16:15:23 +0800

tuntap: fix

Signed-off-by: Jason Wang <jasowang@redhat.com>


---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index bc7cebf..d4ced3a 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -2455,12 +2455,6 @@ static int tun_xdp_one(struct tun_struct *tun,
 		goto out;
 	}
 
-	if (skb_xdp) {
-		err = do_xdp_generic(xdp_prog, skb);
-		if (err != XDP_PASS)
-			goto out;
-	}
-
 	skb_reserve(skb, xdp->data - xdp->data_hard_start);
 	skb_put(skb, xdp->data_end - xdp->data);
 
@@ -2475,6 +2469,12 @@ static int tun_xdp_one(struct tun_struct *tun,
 	skb_reset_network_header(skb);
 	skb_probe_transport_header(skb, 0);
 
+	if (skb_xdp) {
+		err = do_xdp_generic(xdp_prog, skb);
+		if (err != XDP_PASS)
+			goto out;
+	}
+
 	if (!rcu_dereference(tun->steering_prog))
 		rxhash = __skb_get_hash_symmetric(skb);
